src/vivarium/framework/lookup.py

```diff
@@ -1,68 +1,71 @@
 """A set of tools for managing data lookups."""
 from numbers import Number
 from datetime import datetime, timedelta
-import warnings
+from typing import Union, List, Tuple, Callable, TypeVar
 
 import pandas as pd
 
 from vivarium.interpolation import Interpolation
+from vivarium.framework.population import PopulationView
 
+value_type = TypeVar('value_type', Number, timedelta, datetime)
 
-class TableView:
-    def __call__(self, index):
-        raise NotImplementedError()
-
-    def __repr__(self):
-        return "TableView()"
-
-
-class InterpolatedTableView(TableView):
+class InterpolatedTable:
     """A callable that returns the result of an interpolation function over input data.
 
-    Parameters
+    Attributes
     ----------
-    interpolation : callable
-    population_view : `vivarium.framework.population.PopulationView`
-    clock : callable
+    data : `pandas.DataFrame`
```

Comment:
The typing in the signature is sufficient.  The autodoc tools (and more importantly, most editor static analysis tools) pick it up from there.